openedx/core/djangoapps/programs/tasks.py

```diff
@@ -136,89 +162,170 @@ def award_program_certificate(client, user, program_uuid, visible_date):
-    if programs_without_certificates:
-        if str(programs_without_certificates[0]).lower() == "all":
-            # this check will prevent unnecessary logging for partners without program certificates
-            return
-
     countdown = 2**self.request.retries
-    # If the credentials config model is disabled for this
-    # feature, it may indicate a condition where processing of such tasks
-    # has been temporarily disabled.  Since this is a recoverable situation,
-    # mark this task for retry instead of failing it altogether.
 
-    if not CredentialsApiConfig.current().is_learner_issuance_enabled:
+    # If the credentials config model is disabled for this feature, it may indicate a condition where processing of such
+    # tasks has been temporarily disabled.  Since this is a recoverable situation, mark this task for retry instead of
+    # failing it altogether.
+    if not is_credentials_enabled():
         error_msg = (
-            "Task award_program_certificates cannot be executed when credentials issuance is disabled in API config"
+            "Task award_program_certificates cannot be executed, use of the Credentials service is disabled by config"
         )
         LOGGER.warning(error_msg)
         raise _retry_with_custom_exception(username=username, reason=error_msg, countdown=countdown)
 
     try:
-        try:
-            student = User.objects.get(username=username)
-        except User.DoesNotExist:
-            LOGGER.exception(f"Task award_program_certificates was called with invalid username {username}")
-            # Don't retry for this case - just conclude the task.
+        student = User.objects.get(username=username)
+    except User.DoesNotExist:
+        LOGGER.warning(
+            "Task award_program_certificates was called with an invalid username. Could not retrieve a User instance "
+            f"with username {username}"
+        )
+        return
+
+    # this check will prevent unnecessary logging for partners without program certificates
+    programs_without_certificates = configuration_helpers.get_value("programs_without_certificates", [])
+    if programs_without_certificates:
+        if str(programs_without_certificates[0]).lower() == "all":
             return
+
+    LOGGER.info(f"Running task award_program_certificates for user {student}")
+    try:
         completed_programs = {}
         for site in Site.objects.all():
             completed_programs.update(get_completed_programs(site, student))
+
         if not completed_programs:
-            # No reason to continue beyond this point unless/until this
-            # task gets updated to support revocation of program certs.
-            LOGGER.info(f"Task award_program_certificates was called for user {username} with no completed programs")
+            LOGGER.warning(f"Task award_program_certificates was called for user {student} with no completed programs")
```

Comment:
I've also adjusted a number of log levels where it felt appropriate.

Reply:
I'm good with this. 

[musing] It kind of raises the question about whether or not we have an API contract on log levels, you know? I mean, I'm not saying don't do this, but in a complex environment a lot of people will have something that triggers on a specific log message or level. My assumption is that nobody is going to be filtering for warning versus info logs, and we _don't_  have an API contract, for log levels, and I don't think it's common to have one, and I also don't think it's a good idea – developers should feel free to improve log messages without worrying about that kind of message consumption. But in complex environments people do rely about those things. In no way do I think you shouldn't make these changes; I think they are good ones, I'm just pondering.

Reply:
Thank you for including these thoughts. I'll be honest, this hadn't even crossed my mind. However, I know we have, in the past, relied on log messages in other places of the platform to tip off alerts/monitoring. 

I am not aware of any that exists for this specific area of functionality, but I think you're right that I should maybe be a bit more cautious approaching these types of changes in the future. 

Thanks for your insight! I really appreciate it.

Reply:
In hindsight, I think I went a lil overboard. While I was in here, I saw a few log messages the left a bit to be desired (in terms of the information it contained versus information it _could_ contain), and then I started nitpicking log levels and other pieces of it. 

However, I do believe I pushed this in a more helpful direction and these are useful updates.